### ISSUE TITLE
New workspace created with two windows if current window is dedicated

### ISSUE THREAD

**Issue by bmag** [syl20bnr/spacemacs#4838]
#### Description
When the currently selected window is dedicated, creating a new workspace displays two buffers instead of only the home buffer.

#### My two cents
The variable `eyebrowse-new-workspace` is set to `spacemacs/home`, which means that creating a new workspace calls `spacemacs/home` to create the initial window configuration for the workspace. `spacemacs/home` is a bit wrong to use here, because it doesn't check if the current window is dedicated. We need to set `eyebrowse-new-workspace` to a function that really ensures that the home buffer is the only visible buffer. I'll work on it.

#### Reproduction guide
- Start Emacs
- open a buffer (e,g. via the recent files list in the home buffer or via `SPC b b`). You should now have only one window, displaying the buffer.
- dedicate the window to its buffer: `SPC w d`
- create a new workspace: `SPC l w 2`

*Observed behaviour:*
New workspace has two windows: one displaying the buffer, and another one displaying the home buffer.

*Expected behaviour:*
New workspace should have only one window, and it should display the home buffer.

#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.9
- Spacemacs branch: develop (rev. 72e1413)
- Distribution: spacemacs
- Editing style: hybrid
- Completion: ivy
- Layers:
```elisp
(spacemacs-ivy
 (auto-completion :variables auto-completion-enable-snippets-in-popup t auto-completion-enable-help-tooltip t company-tooltip-align-annotations t)
 emacs-lisp git markdown org
 (shell :variables shell-default-height 30 shell-default-position 'bottom)
 syntax-checking version-control ranger evil-cleverparens python command-log eyebrowse smex themes-megapack c-c   ibuffer debug)
```

**Comment by bmag** [comment]
P.S. 1: stumbled into this while trying to debug #4710 
P.S. 2: @StreakyCobra `SPC h I` is nice, thanks

**Comment by StreakyCobra** [comment]
:yellow_heart:

**Comment by bmag** [comment]
Same error happens when creating a new layout (`SPC l 2` instead of `SPC l w 2` in the reproduction guide), I updated the PR accordingly.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
